core/src/main/java/org/apache/carbondata/scan/result/vector/CarbonColumnVector.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.carbondata.scan.result.vector;
+
+import org.apache.spark.sql.types.Decimal;
+
+public interface CarbonColumnVector {
```

Comment:
May we can have offheap implementation in future, but right now the new implementation wrapper class  `ColumnarVectorWrapper`  for passing spark's `ColumnVector` and sets the data to it directly. So I guess it is better to have interface here.